src/common/transformations/src/transformations/common_optimizations/concat_to_broadcast.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/concat_to_broadcast.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/tile.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+static bool use_broadcast(const std::shared_ptr<ov::op::v0::Concat>& concat) {
+    const auto& output = concat->output(0);
+    const auto& input = concat->input(0);
+    const auto& input_concat_dim = input.get_partial_shape()[concat->get_concatenation_axis()];
+
+    return input_concat_dim.is_static() && input_concat_dim.get_length() == 1 && output.get_partial_shape().is_static();
+}
+
+ov::pass::ConcatToBroadcast::ConcatToBroadcast() {
+    MATCHER_SCOPE(ConcatToBroadcast);
+
+    auto concat_label = pattern::wrap_type<op::v0::Concat>([](const Output<Node>& value) {
+        auto node = value.get_node_shared_ptr();
+        if (node->output(0).get_partial_shape().rank().is_dynamic()) {
+            return false;
+        }
+
+        auto first_input_source_output = node->get_input_source_output(0);
+        if (first_input_source_output.get_partial_shape().rank().is_dynamic()) {
+            return false;
+        }
+
+        const auto& input_values = node->input_values();
+
+        return std::all_of(input_values.cbegin(), input_values.cend(), [&](const ov::Output<Node>& output) {
+            return first_input_source_output == output;
+        });
+    });
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto root_node = pattern_map.at(concat_label).get_node_shared_ptr();
+        auto concat = std::dynamic_pointer_cast<op::v0::Concat>(root_node);
+        if (!concat) {
+            return false;
+        }
+
+        if (transformation_callback(concat)) {
+            return false;
+        }
+
+        const auto& input = concat->input_value(0);
+
+        std::shared_ptr<Node> replacement;
+        if (use_broadcast(concat)) {
+            auto target_shape = std::make_shared<ov::op::v0::Constant>(ov::element::i32,
+                                                                       Shape{concat->output(0).get_shape().size()},
+                                                                       concat->output(0).get_shape());
```

Comment:
why do we hard code `target_shape`? If input to Concat even has static shape, we should compute target shape via `ShapeOf` subgraph. The approach with constant target shape is only applicable if input is constant.

Reply:
MOC transformations set all Parameters shapes to dynamic to prevent breaking reshability.
If the shape is static here, it means it doesn't depends on Parameters and we can use it or we are already on plugin stage.

I would agree that ShapeOf approach is slightly better and cover more cases. We will change it in the next PR.

Reply:
Okay, then it is fine. I thought it would be used for MOCTransformations